yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -104,6 +106,7 @@ export class AztecNodeService implements AztecNode {
     protected readonly merkleTreesDb: AztecKVStore,
     private readonly prover: ProverClient | undefined,
     private txValidator: TxValidator,
+    private telemetry: TelemetryClient,
```

Comment:
Just a thought: this reminds me of react prop-drilling. Might it be better for a global like this to have a global getter that returns a "real" or "no-op" client based on environment variables? Then components can request it ad-hoc in their constructors.

Reply:
I had thought about it, but didn't implement it because we don't have other global instances. After drilling the instance in so many places, I think it makes sense to make it global